src/pyradiosky/skymodel.py

```diff
@@ -3055,16 +3077,18 @@ def _select_brightness(
     )
     def select(
         self,
-        component_inds=None,
-        lat_range=None,
-        lon_range=None,
-        min_brightness=None,
-        max_brightness=None,
-        brightness_freq_range=None,
-        inplace=True,
-        run_check=True,
-        check_extra=True,
-        run_check_acceptability=True,
+        component_inds: list[int] | np.ndarray[int] | None = None,
+        lat_range: Latitude | None = None,
+        lon_range: Longitude | None = None,
+        min_brightness: Quantity | None = None,
+        max_brightness: Quantity | None = None,
+        brightness_freq_range: Quantity | None = None,
+        non_nan: Literal["any", "all"] = "all",
```

Comment:
I'm on the fence about whether this should default this way or to None. I think most users would not want to keep sources that have NaN values at all frequencies, but generally select only does some thing you ask for, so this might be surprising to some users.

Reply:
ah right of course! Then in my case I think I agree about default nan handling. Should we be issuing a warning that things are now behaving differently? Also do others have opinions. maybe @steven-murray has an opinion about the nan handling default.

Reply:
On the telecon we agreed to make the warnings point to the select method as a way to fix them. We will default the `non_nan` parameter to `None` but we will by default call select with `non_nan="all"` in read_gleam (adding a keyword to turn it off or set it to `any`).

Reply:
Ok, I made those changes. The only thing I worry about slightly is that it turns out that by defaulting `non_nan="all"` for reading in GLEAM we are also tossing sources with NaN spectral indices when `spectral_type="spectral_index"` because the fitted flux is also NaN in that case. @jpober @mkolopanis are you comfortable with that or should I set the default to `non_nan=None` so that users have to consciously set that parameter?

Reply:
Hmm I could see that being a weird issue for a user. I feel like I am starting to prefer users having to consciously choose parameters when there is no immediately obvious "default" behavior.

Reply:
Yeah, I'm leaning back towards setting the default to None. I guess that also makes me wonder if we want remove the option to `read_gleam_catalog` that I just added since the defaulting would be the same.

Reply:
I don't think I have enough familiarity with this or the context to offer a useful opinion, sorry! In general making users do something explicitly if it might be surprising is a good idea. But of course what is "surprising" might surprise you....

Reply:
I think where I've got it now makes sense. There's no change to default behavior but there are now warnings to let users know about potential issues. And there are options in select to remove components with these issues. There was already some good handling in `at_frequencies` for NaNs, maybe we should think about whether those are highlighted enough.